src/common/token/htoken.go

```diff
@@ -0,0 +1,78 @@
+package token
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"time"
+)
+
+// HToken ...
+type HToken struct {
+	jwt.Token
+}
+
+// NewWithClaims ...
+func NewWithClaims(claims *RobotClaims) *HToken {
```

Comment:
I'd prefer the claims be transparent to the api package so the api code just call NewRobotToken(id, projectID...)